### ISSUE TITLE
Cannot find typings with typescript@beta and @type

### ISSUE THREAD

**Issue by sanex3339** [TypeStrong/ts-loader#247]
My problem same as in https://github.com/TypeStrong/ts-loader/issues/243

After installing @types/node i got multiple errors:

```
(1,21): error TS2307: Cannot find module 'fs'.
(3,23): error TS2307: Cannot find module 'path'.
(80,41): error TS2304: Cannot find name 'process'.
(5,12): error TS2304: Cannot find name 'global'.
(6,5): error TS2304: Cannot find name 'require'.
```

I add all definitions to `types`

But as said https://github.com/mhegazy in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/10113#issuecomment-233199963

> you should not need to add the `types` property, the behavior should be the same with or without it

2.0.0-beta compiler.

project: https://github.com/sanex3339/javascript-obfuscator/tree/dev

project tsconfig:
```json
{
  "compilerOptions": {
    "target": "ES6",
    "module": "commonjs",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "removeComments": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noUnusedLocals": true
  }
}
```

entry point
```ts
"use strict";

import { JavaScriptObfuscator } from './src/JavaScriptObfuscator';

if (!(<any>global)._babelPolyfill) {
    require('babel-polyfill');
}

module.exports = JavaScriptObfuscator;
```

to reproduce you must clone `dev` branch, and run `node_modules/.bin/webpack --watch`

**Comment by sanex3339** [comment]
2.0.0-beta compiler.

project: https://github.com/sanex3339/javascript-obfuscator/tree/dev

project tsconfig:
```json
{
  "compilerOptions": {
    "target": "ES6",
    "module": "commonjs",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "removeComments": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noUnusedLocals": true
  }
}
```

entry point
```ts
"use strict";

import { JavaScriptObfuscator } from './src/JavaScriptObfuscator';

if (!(<any>global)._babelPolyfill) {
    require('babel-polyfill');
}

module.exports = JavaScriptObfuscator;
```

**Comment by brunolm** [comment]
I have the same issue.

I also had issues on VSCode, but they helped me to solve that by doing this: https://github.com/Microsoft/TypeScript/issues/9725#issuecomment-233793570

But this is not enough to solve `ts-loader` issue

**Comment by theduke** [comment]
As specified in #243, adding 

```javascript
  "compilerOptions": {
    ...
    "types": ["react", ...]
  } 
```

to tsconfig.json makes it work, but this should clearly not be neccessary, since `tsc` on itself works fine.

**Comment by gxsjtu** [comment]
any update? 
It is redundant to declare types in tsconfig.

**Comment by blakeembrey** [comment]
I believe this is a fairly trivial fix to do, but I haven't been working on the project as the state of the tests are broken. If someone would care to improve the situation, I can work on new features.

**Comment by HerringtonDarkholme** [comment]
I have a fork to support TypeScript in Vue. I have fixed most tests in TS1.8. https://github.com/HerringtonDarkholme/vue-ts-loader

**Comment by HerringtonDarkholme** [comment]
Yes, I have tried add @types support. Quite easy to implement.

But the tests are really really fragile....

**Comment by blakeembrey** [comment]
Yes, the tests are a pain for me also. The implementation for `ts-node` is https://github.com/TypeStrong/ts-node/commit/240d4b9b751a5c551730a9856502f561844ae3b6, so it shouldn't be hard to add, but with the current state of the tests I'm hesitant to work on it without knowing I won't break something else. Hoping @jbrantly will be able to help with the situation 😄

**Comment by QuantumInformation** [comment]
any ideas how to fix this:

├── ts-loader@0.8.2 
└── UNMET PEER DEPENDENCY webpack@2.1.0-beta.21

**Comment by blakeembrey** [comment]
@QuantumInformation Please don't fill up issues with unrelated problems. If you need help, there's Gitter, Google or a new issue.

Most likely, some package you've installed specifies a peer dependency on `webpack` that is different to the version you have installed.

**Comment by gxsjtu** [comment]
@HerringtonDarkholme could you please make a PR, then the function can be merged to master branch.

**Comment by theduke** [comment]
I figured it out, guys.

It starts working with the following compilerOption settings:

```typescript
"moduleResolution": "node",
"allowSyntheticDefaultImports": true,
```

**Comment by icfantv** [comment]
@theduke setting that flag does not work for us.  this still feels like a bug in ts-loader.

**Comment by marekpw** [comment]
Any news on this? Typescript 2 is no longer in beta and typings are deprecated.

**Comment by HerringtonDarkholme** [comment]
My [fork](https://github.com/HerringtonDarkholme/vue-ts-loader) has added types support, but still cannot get all tests passed. You can also have a look on awesome typescript loader.

**Comment by icfantv** [comment]
We switched to ATL and it works fine.

**Comment by marekpw** [comment]
Can also confirm that ATL works fine, thanks @HerringtonDarkholme for the suggestion.

**Comment by maffelbaffel** [comment]
Same for me. ATL just worked.

**Comment by johnnyreilly** [comment]
Hi @HerringtonDarkholme,

Would you be interested in submitting a PR so we could get types support into ts-loader proper?  We'd be glad to get this issue resolved!

**Comment by johnnyreilly** [comment]
BTW tests should all be passing now.

**Comment by johnnyreilly** [comment]
Hopefully this should be resolved by: https://github.com/TypeStrong/ts-loader/pull/318

**Comment by johnnyreilly** [comment]
Resolved - 🚢 with 0.9.2
